### PR TITLE
ENH: Add version and compose check to workflow

### PR DESCRIPTION
**Pull request by khalford**
Adding an action to check if the `version.txt` is updated correctly not just changed. Also, checks if the `docker-compose.yml` matches `version.txt`

### CODE REVIEW

File: .github/workflows/cloud_chatops.yaml [hunk 1]
```diff
@@ -80,9 +80,41 @@ jobs:
...
+          path: 'main'
+
+      - name: Checkout current working branch
+        uses: actions/checkout@v4
+        with:
+          path: 'branch'
+
+      - name: Compare versions
+        if: ${{ github.ref != 'refs/heads/master' }}
+        uses: khalford/check-version-action@main
```
Comment (DavidFair) [inline .github/workflows/cloud_chatops.yaml:100]: I think this is using an action you maintain? Any reason we can't use a community action / put the steps here?

The concern is the bus factor - if your account got deleted / removed / there was a security update whilst you were away we're unable to update this

**Review by DavidFair — CHANGES_REQUESTED** [review]

File: .github/workflows/cloud_chatops.yaml (hunk 1 above)
Comment (DavidFair) [reply .github/workflows/cloud_chatops.yaml:100]: Could we instead bring either bring the action into `stfc/` or copy the steps into this workflow then?

Comment (khalford) [reply .github/workflows/cloud_chatops.yaml:100]: Yes, a repo will need to be made for the action

**Review by DavidFair — DISMISSED** [review]

File: .github/workflows/cloud_chatops.yaml (hunk 1 above)
Comment (khalford) [reply .github/workflows/cloud_chatops.yaml]: Repo is made and waiting on inital pull request https://github.com/stfc/check-version-action/pull/1

**Comment by khalford** [comment]
Waiting on https://github.com/stfc/check-version-action/pull/1 for the action to be merged. Then this PR can be merged

File: cloud-chatops/version.txt [hunk 2]
```diff
@@ -1 +1 @@
-2.0.1
```
[the remark this answers is not in the document]
Comment (khalford) [reply cloud-chatops/version.txt:1]: I have left them both as the same version. When one of them (either) is merged then the other will be rebased and updated. If I were to change this PR to 2 versions ahead (2.0.3) there would be a chance of a version being skipped.

**Review by meoflynn — APPROVED** [review]

**Comment by khalford** [comment]
Closing due to messy git history when rebasing. Remade into #61

**Comment by khalford** [comment]
Closing due to messy git history when rebasing. Remade into https://github.com/stfc/cloud-docker-images/pull/61

**Pull request closed without merging**